internal/node/hdb/executor.go

```diff
@@ -10,13 +10,18 @@ import (
 // StateUpdate includes all necessary information to update the state of an external system to match
 // the state machine.
 type StateUpdate struct {
-	Index          uint64
-	SchemaType     string
-	DatabaseID     string
-	NewState       []byte
-	Transition     []byte
-	TransitionType string
-	Restore        bool
+	Index      uint64
+	SchemaType string
+	DatabaseID string
+	Restore    bool
+
+	StateUpdateInternal
+}
+
+type StateUpdateInternal interface {
```

Comment:
this confuses me because usually "internal" or "inner" refers to a struct definition while the outer one is the interface, so i think the naming feels kinda backwards. also what was the motivation for this change?

Reply:
Main motivation was to stop having to  unmarshal state bytes in the consumers of the state update, which was getting annoying and making everything harder to read.

I'm gonna try to refactor this to be even simpler.